.github/workflows/frontend.yml

```diff
@@ -70,49 +71,41 @@ jobs:
           echo "::add-matcher::.github/tsc.json"
           echo "::add-matcher::.github/eslint-stylish.json"
 
-      - name: eslint logic
-        id: eslint
-        if: (github.ref == 'refs/heads/master' || needs.files-changed.outputs.eslint_config == 'true' || needs.files-changed.outputs.yarn_lockfile == 'true')
-        run: echo "all-files=true" >> "$GITHUB_OUTPUT"
-
-      # Lint entire frontend if:
-      # - this is on main branch
-      # - eslint configuration in repo has changed
-      # - yarn lockfile has changed (i.e. we bump our eslint config)
-      - name: eslint
-        if: steps.eslint.outputs.all-files == 'true'
-        env:
-          # Run relax config on main branch (and stricter config for changed files)
-          SENTRY_ESLINT_RELAXED: 1
-        run: yarn lint
+      - name: lint
+        if: github.ref == 'refs/heads/master' || needs.files-changed.outputs.lintable_rules_changed != 'true' || needs.files-changed.outputs.lintable_css_in_js_rules_changed != 'true'
```

Comment:
Not sure I understand the logic with running this when rules have not changed

Reply:
ah, i think i have the `!=` wrong here.. it should be like:

IF master OR some rules have changed THEN run everything

although, it could probably just run the set of tasks related to the rules that did change... i can make that update too